superset/migrations/shared/utils.py

```diff
@@ -100,22 +100,31 @@ def paginated_update(
     """
     Update models in small batches so we don't have to load everything in memory.
     """
-    start = 0
-    count = query.count()
+
+    total = query.count()
+    processed = 0
     session: Session = inspect(query).session
+    result = session.execute(query)
+
     if print_page_progress is None or print_page_progress is True:
-        print_page_progress = lambda current, total: print(
-            f"    {current}/{total}", end="\r"
+        print_page_progress = lambda processed, total: print(
+            f"    {processed}/{total}", end="\r"
         )
-    while start < count:
-        end = min(start + batch_size, count)
-        for obj in query[start:end]:
-            yield obj
-            session.merge(obj)
+
+    while True:
```

Comment:
The option was to use either `query.slice(...)` or `session.execute(query).fetchmany(...)`. I opted for the later because:

1. Otherwise one would need to include an `order_by(...)` condition and there's no guarantee that it would be defined.
2. I'm not sure re-executing the query _n_ times using a different OFFSET per query is efficient. Executing the query once and paginating through the result set seems more optimal.